### PR TITLE
fix(client): Prevent the Firefox logo from being displayed on settings pages.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
Add an optional `layoutClassName` property to views.

If a view defines `layoutClassName`, the class name is added to the
`body` element as soon as rendering starts. The class name is
removed in destroy.

issue #3053 

@zaach - r?

### CODE REVIEW

File: app/scripts/views/settings.js [hunk 1]
```diff
@@ -120,9 +120,23 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
       this.logScreen();
     },
 
+    render: function () {
```
Comment (zaach) [inline app/scripts/views/settings.js]: `beforeBeforeRender` :smile:

File: app/scripts/views/settings.js [hunk 2]
```diff
@@ -120,9 +120,23 @@ function ($, modal, Cocktail, Session, BaseView, AvatarMixin,
...
+      // request is made to determine the user's authentication status. On slow
+      // connections this allows the Firefox logo and the white screen used
+      // elsewhere to be displayed while the user's status is being determined.
+      // Adding the `settings` class before the XHR request ensures the Firefox
+      // logo is not displayed. If the user is not authenticated,
+      // `beforeDestroy` will take care of removing the class.
+      // See #3053
+      $('body').addClass('settings');
+
+      return BaseView.prototype.render.apply(this, arguments);
```
Comment (zaach) [inline app/scripts/views/settings.js]: I'm wondering if we should make this a generic feature, like the view could declare a `layout` property (or similar) that `BaseView` adds as a className to `body` early on. Thoughts?

Comment (shane-tomlinson) [inline app/scripts/views/settings.js]: totally. Isn't that the idea of `className`? Where is that written out?

**Comment by shane-tomlinson** [comment]
@zaach - this is ready for a new r?

File: app/tests/spec/views/base.js [hunk 3]
```diff
@@ -316,6 +316,23 @@ function (chai, $, sinon, BaseView, p, Translator, EphemeralMessages, Metrics,
       });
     });
 
+    describe('destroy', function () {
+      it('destroys the view, any subviews, stops listening for messages, and informs others of impending view doom', function () {
```
Comment (vladikoff) [inline app/tests/spec/views/base.js]: nit: `and informs others of impending view doom` wat?

Comment (shane-tomlinson) [inline app/tests/spec/views/base.js]: triggers a `destroy` message.

**Comment by shane-tomlinson** [comment]
@vladikoff - test name updated.

**Pull request merged**